AWSAuthSDK/Sources/AWSMobileClient/AWSUserPoolOperationsHandler.swift

```diff
@@ -66,7 +66,8 @@ AWSCognitoUserPoolInternalDelegate {
     internal static let sharedInstance: UserPoolOperationsHandler = UserPoolOperationsHandler()
 
     static var serviceConfiguration: CognitoServiceConfiguration? = nil
-
+    let completionSourceQueue = DispatchQueue(label: "userpoolhandler.completionsource.queue",
```

Comment:
since you use `.barrier` in each place you use this queue you could leave it as serial.